src/Application.njs

```diff
@@ -16,11 +16,20 @@ class Application extends Nullstack {
     const { value } = await HelloPlugin.sayHello();
   }
 
+  getSubscriptionId() {
+    const monthlyChecked = document.getElementById("monthly").checked
+    if (monthlyChecked) {
+      return 'instill.monthly'
+    }
+    return 'instill.yearly'
+  }
+
   async subscribe() {
+    const productId = this.getSubscriptionId()
     AppSubscriptionPlugin.addListener('onSubscriptionPurchased', (purchase) => {
       this.processSubscription({ purchase: purchase.zzc.nameValuePairs });
     });
-    await AppSubscriptionPlugin.subscribe({ productId: 'google_api' });
+    await AppSubscriptionPlugin.subscribe({ productId });
```

Comment:
In iOS, we are handling some errors that may occur, and we are returning messages to plugin calls, so would be good to add a try-catch here:

```
try {
        await AppSubscriptionPlugin.subscribe({ productId: productId });
} catch (e) {
        console.error(e);
}
```